Source/ablastr/particles/NodalFieldGather.H

```diff
@@ -91,7 +91,31 @@ void compute_weights (const amrex::ParticleReal xp,
     W[1][0] = 1.0_rt - W[1][1];
 
     k = 0;
-#else
+
+#elif defined(WARPX_DIM_RCYLINDER)
+
+    amrex::ignore_unused(zp);
+    const amrex::Real r = (std::sqrt(xp*xp+yp*yp) - plo[0]) * dxi[0] - half_if_cell_centered;
+    i = static_cast<int>(amrex::Math::floor(r));
+    W[0][1] = r - i;
+
+    W[0][0] = 1.0_rt - W[0][1];
+
+    j = 0;
+    k = 0;
+
+#elif defined(WARPX_DIM_RSPHERE)
+
+    const amrex::Real r = (std::sqrt(xp*xp+yp*yp+zp*zp) - plo[0]) * dxi[0] - half_if_cell_centered;
+    i = static_cast<int>(amrex::Math::floor(r));
+    W[0][1] = r - i;
+
+    W[0][0] = 1.0_rt - W[0][1];
+
+    j = 0;
+    k = 0;
+
+#elif (defined WARPX_DIM_1D_Z)
```

Comment:
Maybe this could be rewritten as follows, not sure which one is clearer:
```suggestion

#elif defined(WARPX_DIM_RCYLINDER) || defined(WARPX_DIM_RSPHERE)

#if defined(WARPX_DIM_RCYLINDER)
    amrex::ignore_unused(zp);
    const amrex::Real r = (std::sqrt(xp*xp+yp*yp) - plo[0]) * dxi[0] - half_if_cell_centered;
#elif defined(WARPX_DIM_RSPHERE)
    const amrex::Real r = (std::sqrt(xp*xp+yp*yp+zp*zp) - plo[0]) * dxi[0] - half_if_cell_centered;
#endif

    i = static_cast<int>(amrex::Math::floor(r));
    j = 0;
    k = 0;
    
    W[0][1] = r - i;
    W[0][0] = 1.0_rt - W[0][1];

#elif (defined WARPX_DIM_1D_Z)
```

Reply:
It is clearer to have the code for each dimensionality separated out.